README.md

```diff
@@ -15,6 +47,507 @@ go get -u github.com/src-d/kallax/...
+        Order(kallax.Asc(Schema.User.Username)).
+        Limit(20).
+        Offset(2)
+
+rs, err := store.Find(q)
+if err != nil {
+        // handle error
+}
+
+for rs.Next() {
+        user, err := rs.Get()
+        if err != nil {
+                // handle error
+        }
+}
+```
+
+Next will automatically close the result set when it hits the end. If you have to prematurely exit the iteration you can close it manually with `rs.Close()`.
+
+You can query just a single row with `FindOne`.
+
+```go
+q := NewUserQuery().
+        Where(kallax.Eq(Schema.User.Username, "Joe"))
+
+user, err := store.FindOne(q)
+```
+
+By default, all columns in a row are retrieved. To not retrieve all of them, you can specify the columns to include/exclude.
+
+```go
+// Select only Username and password
+NewUserQuery().Select(Schema.User.Username, Schema.User.Password)
+
+// Select all but password
+NewUserQuery().SelectNot(Schema.User.Password)
+```
+
+### Count results
+
+Instead of passing the query to `Find` or `FindOne`, you can pass it to `Count` to get the number of rows in the resultset.
+
+```go
+n, err := store.Count(q)
+```
+
+### Query with relationships
+
+By default, no relationships are retrieved unless the query specifies so.
+
+For each of your relationships, a method in your query is created to be able to include these relationships in your query.
+
+One to one relationships:
+
+```go
+// Select all posts including the user that posted them
+q := NewPostQuery().WithPoster()
+rs, err := store.Find(q)
+```
+
+One to one relationships are always included in the same query. So, if you have 4 one to one relationships and you want them all, only 1 query will be done, but everything will be retrieved.
+
+One to many relationships:
+
+```go
+// Select all users including their posts
+// NOTE: this is a really bad idea, because all posts will be loaded
+// if the N side of your 1:N relationship is big, consider querying the N store
+// instead of doing this
+// A condition can be passed to the `With{Name}` method to filter the results.
+q := NewUserQuery().WithPosts(nil)
+rs, err := store.Find(q)
+```
+
+To avoid the N+1 problem with 1:N relationships, kallax performs batching in this case. 
+So, a batch of users are retrieved from the database in a single query, then all the posts for those users and finally, they are merged. 
+This process is repeated until there are no more rows in the result.
+Because of this, retrieving 1:N relationships is really fast.
+
+The default batch size is 50, you can change this using the `BatchSize` method all queries have.
+
+**NOTE:** if a filter is passed to a `With{Name}` method we can no longer guarantee that all related objects are there and, therefore, the retrieved records will **not** be writable.
+
+### Reloading a model
+
+If, for example, you have a model that is not writable because you only selected one field you can always reload it and have the full object.
```

Comment:
If you performed any change in an object before calling `reload`, the changes will be overridden with the db retrieved values